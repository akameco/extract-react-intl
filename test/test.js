// @flow weak
const m = require('..')

const pattern = 'test/fixtures/**/*.js'
const locales = ['en', 'ja']

test('extract from file', async () => {
  process.env.BABEL_ENV = 'react-intl'
  const x = await m(locales, pattern, { cwd: './test/fixtures' })
  expect(x).toMatchSnapshot()
})

test('babelrc path resolution', async () => {
  const x = await m(['en'], 'test/resolution/**/*.js', {
    cwd: './test/resolution'
  })
  expect(x).toMatchSnapshot()
})

test('error', async () => {
  expect.assertions(1)
  await m(locales, 'notfound', { cwd: './test/fixtures' }).catch(err => {
    expect(err.message).toMatch('File not found')
  })
})
