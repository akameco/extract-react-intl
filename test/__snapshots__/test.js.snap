// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babelrc path resolution 1`] = `
Object {
  "en": Object {
    "test": "test",
  },
}
`;

exports[`extract from file 1`] = `
Object {
  "en": Object {
    "components.App.hello": "hello",
    "components.App.world": "world",
    "components/Greeting/welcome": "Welcome {name}, you have received {unreadCount, plural, =0 {no new messages} one {{formattedUnreadCount} new message} other {{formattedUnreadCount} new messages}} since {formattedLastLoginTime}.",
  },
  "ja": Object {
    "components.App.hello": "",
    "components.App.world": "",
    "components/Greeting/welcome": "",
  },
}
`;
